plugins/woocommerce/src/Gateways/PayPal/Buttons.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * Class Automattic\WooCommerce\Gateways\PayPal\Buttons file.
+ *
+ * @package WooCommerce\Gateways
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Gateways\PayPal;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_Gateway_Paypal_Request' ) ) {
+	require_once __DIR__ . '/includes/class-wc-gateway-paypal-request.php';
+}
+
+/**
+ * Handles PayPal Buttons.
+ *
+ * @since 10.5.0
+ */
+class Buttons {
+
+	/**
+	 * The option for the client-id.
+	 *
+	 * @var string
+	 */
+	private const CLIENT_ID_OPTION = 'woocommerce_paypal_client_id';
+
+
+	/**
+	 * The gateway instance.
+	 *
+	 * @var \WC_Gateway_Paypal
+	 */
+	private $gateway;
+
+	/**
+	 * Whether the gateway should use Orders v2 API.
+	 *
+	 * @var bool
+	 */
+	private $enabled = false;
+
+	/**
+	 * The request instance.
+	 *
+	 * @var \WC_Gateway_Paypal_Request
+	 */
+	private $request;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param \WC_Gateway_Paypal $gateway The gateway instance.
+	 */
+	public function __construct( \WC_Gateway_Paypal $gateway ) {
+		$this->gateway = $gateway;
+		$this->request = new \WC_Gateway_Paypal_Request( $this->gateway );
+
+		// phpcs:ignore Generic.Commenting.Todo.TaskFound
```

Comment:
Nit: I think we can remove this ignore -- it looks like it's ignoring a TODO comment, but that comment doesn't exist in this code.
```suggestion
```